fabric-enchanting-api-v1/build.gradle

```diff
@@ -0,0 +1,6 @@
+archivesBaseName = "fabric-enchanting-api-v1"
```

Comment:
Well it's time to ask the question. What future api could be related to enchantments/enchanting is feasible that could belong here? As this is a reasonably small module (two interfaces in api) I don't know if this is a micromodule in the typical sense.

The enchanting target workaround is quite explicit in it's purpose. It *may* work fine in the `item-api` as enchantments are applied directly to items (maybe in an enchanting subpackage but it definitely sounds a bit odd.

The enchanting power logic seems to be firmly in the territory of a `block-extensions-api` which I theorize would also have logic for stuff like dynamic slipperiness, more dynamic climbing logic and other stuff vanilla hardcodes away in relation to blocks.

Reply:
I don't know what's the current concept for fapi. I thought it was to have a lot of stuff neatly modularized.
Do we do bigger modules now?

Reply:
I don't think the standard has changed too much. Obviously we want to be as modular as possible. However some stuff like **7** rendering related modules is kinda annoying.

Some stuff like object builders (fabric-object-builder-api-v1) and lifecycle events (fabric-lifecycle-events-v1) are larger modules by virtue of covering more than a few things like the object builder module being a general catch-all for exposing protected/private things in minecraft codebase related to creation of objects. The lifecycle events are larger due to covering both client, server and some chunk/entity/block entity related lifecycle events.

Enchantments may be niche enough to where it is merited it's own module but I did want to gauge how much more growth the enchantments module could see in the decision of keeping it it's own module or integrating it into a more general new module and or some existing modules.

Reply:
I understand where you go with the block-extensions - that makes kind of sense.
But the item-api doesn't really fit imo since it only consists of an event right now and the enchanting process doesn't need events.
Keeping just the enchanting process in its own module would be kind of strange too because that's honestly a bit little for a module that probably won't grow that much.